XenAdmin/TabPages/ManageUpdatesPage.cs

```diff
@@ -400,14 +399,50 @@ private void UpdateDetails()
 
         }
 
-        private static string RecommendedPatchesForHost(Host host)
+        private static string RequiredUpdatesForHost(Host host)
         {
-            var result = new List<string>();
-            var recommendedPatches = Updates.RecommendedPatchesForHost(host);
+            var requiredUpdates = Updates.RecommendedPatchesForHost(host);
 
-            foreach (var patch in recommendedPatches)
+            if (requiredUpdates == null)
+            {
+                // versions with no minimum patches
+                var updatesList = new List<string>();
+                var alerts = new List<Alert>(Updates.UpdateAlerts);
+                if (alerts.Count == 0)
+                    return String.Empty;
+
+                foreach (Alert alert in alerts)
+                {
+                    if (!(alert is XenServerPatchAlert))
+                        continue;
+                    if (alert.AppliesTo.Contains(host.Name))
+                        updatesList.Add(alert.Name);
+                }
+
+                updatesList.Sort(StringUtility.NaturalCompare);
+                return string.Join(", ", updatesList.ToArray());
+            }
+
+            else
+            {
+                // versions with minimum patches
+                var result = new List<string>();
+                foreach (var patch in requiredUpdates)
+                    result.Add(patch.Name);
+
+                result.Sort(StringUtility.NaturalCompare);

```

Comment:
Shouldn't we show the minimal patches in the original order, as this is the order that they need to be applied on? ~Also, on the General tab we show them in the original order...

Reply:
I see this as just a list of the missing ones. We could use the minimal patch order, but because for older hosts we don't know that, sorting wouldn't be consistent between old and new pools.

If that's what we'd like to have, I'm fine with it, despite I still prefer the alphabetical sorting as I see this rather as a list, not as an ordered install sequence.

Reply:
We're already using the minimal list instead of the full list for newer pools, so they're already not consistent. I think we should preserve consistency with the general tab.